operator/api/v1beta1/helmrelease_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// HelmReleaseSpec defines the desired state of HelmRelease
+type HelmReleaseSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	ReleaseName string               `json:"releaseName"`
+	Chart       ChartSpec            `json:"chart"`
+	Values      runtime.RawExtension `json:"values"`
```

Comment:
So in order to get around not being able to use map[string]interface{} to autogen a CRD, I was able to use this type from `runtime`. This will allow us to continue to use kubebuilder to autogen the CRD. _However_ this means that we need to Unmarshal the contents of this field into a map when we want to use it its contents.

Reply:
I'm aware that this is not exactly the right type to use for this, there was also a `JSON` type but it threw errors because (ironically) that type is missing JSON struct tags. 

I would rather do it this way initially than introduce a toolchain to this project that's broken from the get-go. It wont be difficult to go back and remove the JSON unmarshalling code when we're able to change this type to `map[string]interface{}`.

Reply:
Using `[]byte` will generate a CRD that looks like this:

```
values:
   format: byte
   type: string
```

Which will mean we'd need to pipe YAML or JSON in as a string